lib/groupdate/series_builder.rb

```diff
@@ -186,7 +186,10 @@ def generate_series(data, multiple_groups, group_index)
           last_step = series.last
           loop do
             next_step = last_step + step
-            next_step = round_time(next_step) if next_step.hour != day_start # add condition to speed up
+
+            if next_step.hour.seconds != day_start # add condition to speed up
```

Comment:
@ankane day_start here is time in seconds, `next_step.hour.seconds != day_start` better here.